src/tip-opengov.ts

```diff
@@ -92,48 +100,40 @@ async function signAndSendCallback(
   bot: Probot,
   contributor: ContributorAccount,
   type: "preimage" | "referendum",
-  unsubscribe: () => void,
-  result: ISubmittableResult,
+  result: TxFinalizedPayload,
 ): Promise<ExtrinsicResult> {
   return await new Promise((resolve, reject) => {
-    const resolveSuccess = (blockHash: string) => {
-      unsubscribe();
-      resolve({ success: true, blockHash });
-    };
-    if (result.status.isInBlock) {
-      const blockHash = result.status.asInBlock.toString();
+    if (result.ok) {
+      const blockHash = result.block.hash;
       bot.log(`${type} for ${contributor.address} included at blockHash ${blockHash}`);
       if (process.env.NODE_ENV === "test") {
         // Don't have to wait for block finalization in a test environment.
-        resolveSuccess(blockHash);
+        resolve({ success: true, blockHash });
       }
-    } else if (result.status.isFinalized) {
-      const blockHash = result.status.asFinalized.toString();
-      bot.log(`Tip for ${contributor.address} ${type} finalized at blockHash ${blockHash}`);
-      resolveSuccess(blockHash);
-    } else if (result.isError) {
-      bot.log(`status to string`, result.status.toString());
-      bot.log(`result.toHuman`, result.toHuman());
+    } else if (!result.ok) {
```

Comment:
Isn't `result.ok` always defined, and `boolean`? Sounds like `else` clause won't be reachable